test/test_nn.py

```diff
@@ -6936,11 +6965,25 @@ def get_grid(device='cpu', data=None):
                                                         align_corners=align_corners),
                         (input, grid)))
 
+                    F.grid_sample(input, grid, mode=mode, padding_mode=padding_mode, align_corners=align_corners).sum().backward()
+                    input_grad, grid_grad = input.grad, grid.grad
+                    input.grad, grid.grad = None, None
+
+                    torch._grid_sampler_2d_cpu_fallback(
+                        input.float(), grid.float(),
+                        F.GRID_SAMPLE_INTERPOLATION_MODES[mode],
+                        F.GRID_SAMPLE_PADDING_MODES[padding_mode],
+                        align_corners).sum().backward()
+                    # Only compare up to float precision
+                    self.assertEqual(input.grad.float(), input_grad.float())
+                    self.assertEqual(grid.grad.float(), grid_grad.float())
+
```

Comment:
We already compare the gradients in `test`, I don't think there is a need to do that again here. My guess is that `gradcheck` adds some extra tests, but what concretely I don't know

Reply:
My understanding is that `gradcheck` compares against a numerical jacobian calculated from the change in the outputs after taking small "steps" in the input space. It doesn't work with the fallback here because it requires double precision to make those small steps.